### PR TITLE
Eventvolunteercancel

### PR DESCRIPTION
**Pull request by JakeRoeleven**

### CODE REVIEW

File: app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java [hunk 1]
```diff
@@ -70,13 +106,70 @@ protected void onCreate(Bundle savedInstanceState) {
...
+//
+//                }
+//            }
+//
+//            @Override
+//            public void onCancelled(@NotNull DatabaseError databaseError) {
+//
+//            }
+//        });
+//        ////////////////////////////////////////////////////////
```
Comment (Jahmilli) [inline app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java]: do you need this comment for later?

File: app/src/main/res/menu/drawer_view_charity.xml [hunk 2]
```diff
@@ -7,8 +7,10 @@
         <item
             android:id="@+id/nav_profile"
             android:title="Profile" />
+    </group>
+    <group android:checkableBehavior="single">
         <item
             android:id="@+id/nav_logout"
-            android:title="Log Out Charity" />
+            android:title="Log Out Volunteer" />
```
Comment (Jahmilli) [inline app/src/main/res/menu/drawer_view_charity.xml]: why was this changed? This is the charity drawer.

File: app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java [hunk 3]
```diff
@@ -70,13 +106,70 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        reference.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NotNull DataSnapshot dataSnapshot) {
+                if (dataSnapshot.child("Events").child(eventID).getValue() != null) {
+                    eventRegisterButton.setVisibility(View.GONE);
+                    if (dataSnapshot.child("Events").child(eventID).getValue().toString().equals("pending") || dataSnapshot.child("Events").child(eventID).getValue().toString().equals("registered")) {
+                        eventRegisterStatusTV.setText("You have registered for this event");
+                        eventCancelButton.setVisibility(View.VISIBLE);
+                    }
+                    if (dataSnapshot.child("Events").child(eventID).getValue().toString().equals("cancelled")) {
```
Comment (Jahmilli) [inline app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java]: rather than doing `dataSnapshot.child("Events").child(eventID)` every time here, maybe just go to the child from reference.
ie:
`reference.child("Events").child(eventID).addListener.......`
then you can reference to event by just doing `datasnapshot.getValue()`

File: app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java [hunk 4]
```diff
@@ -53,6 +74,21 @@ protected void onCreate(Bundle savedInstanceState) {
         Event event = intent.getParcelableExtra("event");
 
         eventID = event.getId();
+        charityID = event.getOrganisers();
+        nReference = database.getReference("Charities").child(charityID);
```
Comment (Jahmilli) [inline app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java:78]: if nReference is referring to the particular charity, could we rename it to like `charityReference` or something?

File: app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java [hunk 5]
```diff
@@ -70,13 +106,70 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            public void onDataChange(@NotNull DataSnapshot dataSnapshot) {
+                if (dataSnapshot.child("Events").child(eventID).getValue() != null) {
+                    eventRegisterButton.setVisibility(View.GONE);
+                    if (dataSnapshot.child("Events").child(eventID).getValue().toString().equals("pending") || dataSnapshot.child("Events").child(eventID).getValue().toString().equals("registered")) {
+                        eventRegisterStatusTV.setText("You have registered for this event");
+                        eventCancelButton.setVisibility(View.VISIBLE);
+                    }
+                    if (dataSnapshot.child("Events").child(eventID).getValue().toString().equals("cancelled")) {
+                        eventRegisterStatusTV.setText("You have cancelled your registration for this event");
+                        eventCancelButton.setVisibility(View.GONE);
```
Comment (Jahmilli) [inline app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java:123]: I think you already set this button to 'GONE' in the xml. Do you need to set it again here?

File: app/src/main/res/layout/activity_event_register.xml [hunk 6]
```diff
@@ -35,6 +36,36 @@
...
+
+                <TextView
+                    android:id="@+id/eventRegisterEventHost"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="20dp"
+                    android:text="Event Hosted By:  " />
+
+                <TextView
+                    android:id="@+id/eventRegisterEventHostTV"
```
Comment (Jahmilli) [inline app/src/main/res/layout/activity_event_register.xml:62]: Can we underline this or something, idk the best way to show this but its not obvious that clicking on the name goes to the charity details atm. 
Once Jacob fixes up the design we can adjust it, but just so everyone knows thats a thing it would be good to make the fact that this can direct to another page more apparent.

**Review by Jahmilli — CHANGES_REQUESTED** [review]
Change is great, just wondering if we want to commit to not allowing users to reregister for events if they cancel. Not sure if there is any benefit to doing this but maybe there is a good reason, just let me know.

I've added some comments on some stuff that may need to be fixed up.

File: app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java (hunk 5 above)
Comment (JakeRoeleven) [reply app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java:123]: It is set to gone then set to visible then set to gone again so you can't cancel again. In latest commit set register button visible again so can reregister

File: app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java (hunk 3 above)
Comment (JakeRoeleven) [reply app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java]: Fixed with latest commit

File: app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java (hunk 1 above)
Comment (JakeRoeleven) [reply app/src/main/java/team7/voluntime/Activities/EventRegisterActivity.java]: Was going to use it to try and show registration status of event list adapter but not going to work that way so fixed in the latest commit

File: app/src/main/res/layout/activity_event_register.xml (hunk 6 above)
Comment (JakeRoeleven) [reply app/src/main/res/layout/activity_event_register.xml:62]: Done in latest commit

File: app/src/main/res/menu/drawer_view_charity.xml (hunk 2 above)
Comment (JakeRoeleven) [reply app/src/main/res/menu/drawer_view_charity.xml]: Oh whoops wanted to try and find a way to lock log out to the button of the drawer because I kept logging myself out. Fixed

**Review by JakeRoeleven — COMMENTED** [review]
Changes Completed

**Review by Jahmilli — APPROVED** [review]
lgtm
